kong/router/compat.lua

```diff
@@ -133,8 +134,7 @@ local function get_expression(route)
     if op == OP_REGEX then
       -- 1. strip leading `~`
       -- 2. prefix with `^` to match the anchored behavior of the traditional router
-      -- 3. update named capture opening tag for rust regex::Regex compatibility
-      return "^" .. p:sub(2):gsub("?<", "?P<")
+      return sanitize_regex("^" .. p:sub(2))
```

Comment:
What do you think of moving _all_ the string modifications into a single function?

```suggestion
      return prepare_path_regex(p)
```

Reply:
This is for compatibility with PCRE, all things to do are string substitute.